src/network.rs

```diff
@@ -1,3 +1,39 @@
 //! Network specific utils
 
+use std::fmt::Debug;
+
+use alloy_primitives::U256;
+use alloy_rpc_types::{Header, Transaction};
+use revm::primitives::{BlockEnv, SpecId, TxEnv};
+
+use crate::mv_memory::MvMemory;
+
+/// A chain ID (u64) associated with relevant utils.
```

Comment:
A trait doesn't define the full structure that implements it. A custom network may hold much more data than an id to, for example, build more efficient `MvMemory`. Something like `Custom behaviours for different chains & networks` would be better.